### PR TITLE
fix(build): relup for otp24 failed

### PR DESCRIPTION
**Pull request by terry-xiaoyu**
The defination of `#application{}` record in systools.hrl is changed in OTP 24.
So we need a rebar3 binary compiled with OTP 24.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 $(shell $(CURDIR)/scripts/git-hooks-init.sh)
-REBAR_VERSION = 3.14.3-emqx-8
+REBAR_VERSION = 3.18.0-emqx-1
```
Comment (zmstone) [inline Makefile:2]: move this version to `ensure-rebar3`

File: scripts/ensure-rebar3.sh [hunk 2]
```diff
@@ -3,14 +3,19 @@
...
+REBAR3_FILENAME="${REBAR3_FILENAME:-rebar3}"
 
 # ensure dir
 cd -P -- "$(dirname -- "${BASH_SOURCE[0]}")/.."
 
 DOWNLOAD_URL='https://github.com/emqx/rebar3/releases/download'
 
 download() {
-    curl -f -L "${DOWNLOAD_URL}/${VERSION}/rebar3" -o ./rebar3
+    curl -f -L "${DOWNLOAD_URL}/${VERSION}/${REBAR3_FILENAME}" -o ./rebar3
```
Comment (zmstone) [inline scripts/ensure-rebar3.sh:14]: download old version (`3.14.3-emqx-8`) for otp 23 and new version for 24 ?

Comment (terry-xiaoyu) [reply scripts/ensure-rebar3.sh:14]: OK, I'll change this.

File: Makefile (hunk 1 above)
Comment (terry-xiaoyu) [reply Makefile]: changed

**Review by tigercl — APPROVED** [review]

**Review by DDDHuang — APPROVED** [review]

**Pull request merged**